### PR TITLE
Autoencoders

### PR DESCRIPTION
**Pull request by vaibhkrishan**
@KanwalSingh please review for code quality and documentation etc.

### CODE REVIEW

File: autoencoders/autoencoder.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        rbms = []
+        for i in range(num_layers - 1):
+            rbms.append(RBM(layer_units[i], layer_units[i+1]))
+
+        self.layer_units = layer_units
+        self.num_layers = num_layers
+        self.rbms = rbms
+        self.ann = neural_net(layer_units + list(reversed(layer_units))[1:])
+
+    def train_as_rbm(self, data, max_epoch = 50001, threshold = 0.005,
```
Comment (KanwalSingh) [inline autoencoders/autoencoder.py]: @vaibhkrishan the documentation went inside the function, correct this

File: autoencoders/autoencoder.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                            threshold = threshold,
+                            grad_threshold = grad_threshold,
+                            learning_rate = learning_rate)
+            layer_data = layer_rbm.run_visible(layer_data)
+
+    def run_rbms(self, data, give_probs = False):
+        """
+        Runs all the RBMs forward and backward to get outputs.
+
+        @type   data:       numpy array
```
Comment (KanwalSingh) [inline autoencoders/autoencoder.py]: @vaibhkrishan correct the documentation - numpy array to numpy matrix everywhere (or based on the case)

File: autoencoders/neural_net.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
+import numpy
+import numpy.linalg
+import scipy.special
+
+def KL_term(sparsity, target_sparsity):
```
Comment (KanwalSingh) [inline autoencoders/neural_net.py]: @vaibhkrishan handle the NAN case in sparsity metric

File: LICENSE.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2015 IREF(https://www.indianrealestateforum.com)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
```
Comment (KanwalSingh) [inline LICENSE.md]: @vaibhkrishan Add the author name and reviewer's name

File: autoencoders/rbm.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
+import numpy
```
Comment (KanwalSingh) [inline autoencoders/rbm.py]: @vaibhkrishan take the rbm.py into a new module of rbm. 
Do the same for others as well , neural_net , or CNN / RNN  (in future)

**Comment by vaibhkrishan** [comment]
@KanwalSingh , made changes as per comments. Review and tell me whether to merge.

**Comment by KanwalSingh** [comment]
@vaibhkrishan as the next steps we should add some examples to using this code.
